server: refactor: introduce `AuthUser` interface

paul-marechal opened the pull request:
Replace the references to `OAuth2User` by `AuthUser`. This allows downstream extenders to more easily contribute alternative OAuth2 providers: If the expected data is stored in different attributes it will be possible to bridge it by implementing the proper `AuthUser`.

New configuration example:

```yml
ovsx:
  oauth2:
    provider: gitlab
    attribute-names:
      gitlab:
        avatar-url: avatar_url
        email: email
        full-name: name
        login-name: username
        provider-url: '' # idk what this is for

spring:
  security:
    oauth2:
      client:
        registration:
          gitlab:
            client-id: [client-id]
            client-secret: [client-secret]
            authorization-grant-type: authorization_code
            redirect-uri: http://[openvsx-host-name]/login/oauth2/code/gitlab
            scope: read_user
            clientName: OpenVSX
        provider:
          gitlab:
            authorization-uri: http://[gitlab-host-name]/oauth/authorize
            token-uri: http://[gitlab-host-name]/oauth/token
            user-info-uri: http://[gitlab-host-name]/api/v4/user
            jwk-set-uri: http://[gitlab-host-name]/oauth/discovery/keys
            user-name-attribute: id
```

paul-marechal commented:
Note that there seems to be an issue with formatting in this project...

I'm using VS Code along its Java extension to develop and it often feels like reorganizing or cleaning up imports.

I'd be in favor of keeping the cleanup as part of these changes.

It would be good to look into making auto-formatting part of the build step, maybe as a gradle plugin of some kind?

HeikoBoettger commented on server/src/main/java/org/eclipse/openvsx/security/DefaultAuthUser.java:
```diff
@@ -0,0 +1,68 @@
+        avatarUrl = oauth2User.getAttribute("avatar_url");
+        email = oauth2User.getAttribute("email");
+        fullName = oauth2User.getAttribute("name");
+        loginName = oauth2User.getAttribute("login");
```
After looking closer at this. I wonder what's the difference between `loginName` and `authId` on GitHub. I suggest to name this class GitHubAuthUser.

paul-marechal replied:
> After looking closer at this. I wonder what's the difference between `loginName` and `authId` on GitHub.

The naming of attributes like `oauth2User.getAttribute("login");` is completely arbitrary on the OAuth2 provider side. GitLab's `authId` might be GitHub's `login` equivalent just because.

> If the expected data is stored in different attributes it will be possible to bridge it by implementing the proper `AuthUser`.

This was my original idea with this PR: the code would no longer hardcode attribute names in several locations: Attribute access would be abstracted in one place: within `AuthUser` implementations.

> I wonder whether in a more generic environment one should use whatever is set via the `spring.security.oauth2.client.provider.[id[.user-name-attribute` configuration value instead of using the hardcoded name "login"

It's not a bad idea, I'll see what I can do about it later.

paul-marechal replied:
Looked into it and I'd like to keep the PR as is for now.

My main issue is how rigid the code currently is: The `OAuth2UserServices` hardcodes the expect registration IDs such as `github` and `eclipse`. In our case we just override components to link to an alternative OAuth2 provider so we're skipping that problematic logic from OpenVSX. This `AuthUser` abstraction is enough for us to then propagate information down the rest of the system.

Feel free to contribute a more generic authentication system later.

paul-marechal commented:
Rebased my branch.

paul-marechal commented on server/src/main/java/org/eclipse/openvsx/security/AuthUserFactory.java:
```diff
@@ -0,0 +1,20 @@
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AuthUserFactory {
+
+    public AuthUser createAuthUser(String providerId, OAuth2User oauth2User) {
+        return new DefaultAuthUser(
+            oauth2User.getName(),
+            oauth2User.getAttribute("avatar_url"),
+            oauth2User.getAttribute("email"),
+            oauth2User.getAttribute("name"),
+            oauth2User.getAttribute("login"),
+            providerId,
+            oauth2User.getAttribute("html_url")
+        );
+    }
+}
```
@HeikoBoettger this class should make it easier to customize `AuthUser` instances. Right now it is hardcoded but you can later make this factory depend on some Spring configuration to customize attribute access.

HeikoBoettger replied:
@paul-marechal I am not so familiar with spring, do you know how to map the "login" in the spring configuration? I only know about the "spring.security.oauth2.client.provider.my-oauth-provider.user-name-attribute=name". May be short hint somewhere in the openvsx documentation on how to customize the attribute access would be good.

Great work by the way.

amvanbaren commented:
@HeikoBoettger Can you share which additional identity provider you were able to implement using this PR?

HeikoBoettger-KarlStorz commented:
> @HeikoBoettger Can you share which additional identity provider you were able to implement using this PR?

@amvanbaren Somehow I lost my comment. The PR didn't help completely, what is missing is basically a mapping of the login-attribute. One could add a GitLab specific mapping to the code, what I did based on this merge request adding a completely new generic provider type which uses additional setting from the application.yaml file defining for the web-interface what provider to use for the login-link and how to map the login (supporting mapping any of the attributes in the AuthUserFactory).

If the code could be changed to use spring.security.oauth2.client.provider.[providerId].user-name-attribute / providerDetails.userInfoEndpoint.userNameAttributeName as described in https://docs.spring.io/spring-security/reference/servlet/oauth2/login/core.html#oauth2login-sample-application-config. Instead of login I could abuse the GitHub provider entry.

To fully support other providers the UI would require a change to select the login method and the AuthToken inside the UserData class need to be contain a map from provider to token: https://github.com/eclipse/openvsx/blob/072bf3e7fa38eef217ebb30d9748f5556440c665/server/src/main/java/org/eclipse/openvsx/entities/UserData.java#L63

Or the code gets simplified to just store an AuthToken for one provider at a time. I don't know why there are two fields in the UserData when there is only one provider-String.

HeikoBoettger-KarlStorz commented:
@paul-marechal Is it possible to rebase this branch on to the latest master?

amvanbaren commented:
> Or the code gets simplified to just store an AuthToken for one provider at a time. I don't know why there are two fields in the UserData when there is only one provider-String.

The `githubToken` is to login. The `eclipseToken` is to check the publisher agreement. 
We could rename `githubToken` to `loginToken`.

paul-marechal commented:
Ok so I initially didn't want to bother implementing the configuration of attributes because it adds a decent amount of complexity... I changed my mind figuring it might make someone else's life easier.

The configuration may look like this:

```yml
ovsx:
  auth:
    attribute-names:
      gitlab:
        avatar-url: awdawd
        email: awdawd
        full-name: awdawd
        login-name: awdawd
        provider-url: awdawd
```

amvanbaren commented:
@paul-marechal Thanks for providing the configuration. I'll try to configure Google OAuth using this PR.

paul-marechal commented:
I want to apologize for forgetting to transpose code to handle `OidcUserRequest` we had to use for our fork... Without it we couldn't access certain attributes.

Second, there's still a need to manually edit the webui code to change the OAuth2 authorization endpoint used from `github` to whatever you want to reach. I'm not sure if we can somehow get rid of it or make it configurable yet. I will try so it's also less of a hassle...

amvanbaren commented:
> there's still a need to manually edit the webui code

Yes, the login url is hardcoded in the webui:
https://github.com/eclipse/openvsx/blob/0f5315df6d38e2a983cd3aa72d76215626039bd8/webui/src/extension-registry-service.ts#L27-L29

The `UserAPI` has a login endpoint, but it causes CORS errors:
https://github.com/eclipse/openvsx/blob/0f5315df6d38e2a983cd3aa72d76215626039bd8/server/src/main/java/org/eclipse/openvsx/UserAPI.java#L72-L77

paul-marechal commented:
I implemented the dynamic redirection based on a server configuration, I didn't encounter a CORS error?

```yml
ovsx:
  auth:
    provider: azure
```

paul-marechal commented:
Using the latest commit I can now assert that no more code changes are required to configure a custom provider (following our use cases) as I was able to completely configure authentication to Azure using only the server spring config. I initially forgot to include some code changes...

HeikoBoettger-KarlStorz commented:
@paul-marechal  Just to give an early feedback. I compared your changes with the modifications I did on my local branch I was thinking of contributing and it's more or less the same.

One thing I wonder is, do you use the githubToken to store the token in the database (UserData)? The TokenService also contains a switch block with the providers.

paul-marechal commented:
I didn't find it required to store anything in `githubToken`.

paul-marechal commented on server/src/main/java/org/eclipse/openvsx/security/OAuth2UserServices.java:
```diff
@@ -100,27 +95,34 @@ public void authenticationSucceeded(AuthenticationSuccessEvent event) {
-                return loadEclipseUser(userRequest);
-            default:
-                throw new CodedAuthException("Unsupported registration: " + registrationId, UNSUPPORTED_REGISTRATION);
+        if (registrationId == "eclipse") {
```
This is bogus, I forgot Java doesn't know how to compare string equality...

HeikoBoettger-KarlStorz commented:
Note sure whether I did something wrong but I get an error message: "Missing configuration: ovsx.auth.attribute-names.gitlab".
Shouldn't there be an default value.

HeikoBoettger-KarlStorz commented:
@paul-marechal First of all thank you for this awesome work.

> ```yaml
> ovsx:
>   auth:
>     attribute-names:
>       gitlab:
>         avatar-url: awdawd
>         email: awdawd
>         full-name: awdawd
>         login-name: awdawd
>         provider-url: awdawd
> ```

I used the following values for gitlab:

```yaml
ovsx:
  auth:
    attribute-names:
      gitlab:
        avatar-url: avatar_url
        email: email
        full-name: name
        login-name: username
```

I don't know what to set for the `provider-url` on gitlab but with this mapping I can confirm that I am able to login.

For `spring.security.oauth2` I used the following settings:

```text
spring.security.oauth2.client.registration.gitlab.client-id=[client-id]" 
spring.security.oauth2.client.registration.gitlab.client-secret=[client-secret]
spring.security.oauth2.client.registration.gitlab.authorization-grant-type=authorization_code
spring.security.oauth2.client.registration.gitlab.redirect-uri=http://[gitlab-host-name]/login/oauth2/code/gitlab
spring.security.oauth2.client.registration.gitlab.scope=read_user
spring.security.oauth2.client.registration.gitlab.clientName=OpenVSX
spring.security.oauth2.client.provider.gitlab.authorization-uri=http://[gitlab-host-name]/oauth/authorize
spring.security.oauth2.client.provider.gitlab.token-uri=http://[gitlab-host-name]/oauth/token
spring.security.oauth2.client.provider.gitlab.user-info-uri=http://[gitlab-host-name]/api/v4/user
spring.security.oauth2.client.provider.gitlab.jwk-set-uri=http://[gitlab-host-name]/oauth/discovery/keys
spring.security.oauth2.client.provider.gitlab.user-name-attribute=id
```

HeikoBoettger-KarlStorz commented:
For completeness this is the configuration inside `gitlab`:
![image](https://github.com/user-attachments/assets/bd280392-267d-47e7-bf78-674216897968)

paul-marechal commented:
I noticed that there's another PR opened which allows to auth users without oauth2, so I renamed the configuration I introduced from `ovsx.auth` to `ovsx.oauth2` so it better match its intended purpose.

I also updated the main PR description with a configuration example I took from @HeikoBoettger-KarlStorz :)

HeikoBoettger-KarlStorz commented:
@amvanbaren Anything still missing to get this released? I can confirm this is working with both microsoft entra/azure sso (using a corporate tenant) and gitlab.

amvanbaren commented:
I think this feature is ready to go.
@HeikoBoettger-KarlStorz Can you do a final test with both microsoft entra/azure sso (using a corporate tenant) and gitlab?

HeikoBoettger-KarlStorz commented:
@amvanbaren Yes, I will test it tomorrow morning.

HeikoBoettger-KarlStorz commented:
@amvanbaren Do you some more information on what has changed according to the /login end-point. I see that ther was a change to /login-providers but I that end-point now says that there are no login providers.

HeikoBoettger-KarlStorz commented:
I used in the generate-properties.sh for testing, but I the gitlab config isn't read into the attributeConfigs:
```
echo "ovsx.auth2.attribute-names.gitlab.avatar-url=avatar_url" >> $OVSX_APP_PROFILE
echo "ovsx.auth2.attribute-names.gitlab.email=email" >> $OVSX_APP_PROFILE
echo "ovsx.auth2.attribute-names.gitlab.full-name=name" >> $OVSX_APP_PROFILE
echo "ovsx.auth2.attribute-names.gitlab.login-name=username" >> $OVSX_APP_PROFILE
echo "ovsx.auth2.attribute-names.gitlab.provider-url=html_url" >> $OVSX_APP_PROFILE
```
Any guess what I am doing wrong?

HeikoBoettger-KarlStorz commented:
> I used in the generate-properties.sh for testing, but I the gitlab config isn't read into the attributeConfigs:
> 
> ```
> echo "ovsx.auth2.attribute-names.gitlab.avatar-url=avatar_url" >> $OVSX_APP_PROFILE
> echo "ovsx.auth2.attribute-names.gitlab.email=email" >> $OVSX_APP_PROFILE
> echo "ovsx.auth2.attribute-names.gitlab.full-name=name" >> $OVSX_APP_PROFILE
> echo "ovsx.auth2.attribute-names.gitlab.login-name=username" >> $OVSX_APP_PROFILE
> echo "ovsx.auth2.attribute-names.gitlab.provider-url=html_url" >> $OVSX_APP_PROFILE
> ```
> 
> Any guess what I am doing wrong?

Missed the `o`, must be `ovsx.oauth2` but I still get an error that no authentication request is found when the gitlab server invokes the `login/oauth2/code/gitlab` end-point.

HeikoBoettger-KarlStorz commented:
@amvanbaren I haven't tried with multiple providers at once due to other tasks on my end. If needed I will do so as well. Please let me know.

nfogh commented:
Hi. I'm trying to get this branch to work for a local deployment (very excited about it - thanks for the work). But I'm getting an error when launching the server:

`
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'extensionService' defined in file [/app/build/classes/java/main/org/eclipse/openvsx/ExtensionService.class]: Unsatisfied dependency expressed through constructor parameter 3: Error creating bean with name 'publishExtensionVersionHandler' defined in file [/app/build/classes/java/main/org/eclipse/openvsx/publish/PublishExtensionVersionHandler.class]: Unsatisfied dependency expressed through constructor parameter 5: Error creating bean with name 'userService' defined in file [/app/build/classes/java/main/org/eclipse/openvsx/UserService.class]: Unsatisfied dependency expressed through constructor parameter 5: Error creating bean with name 'clientRegistrationRepository' defined in class path resource [org/springframework/boot/autoconfigure/security/oauth2/client/servlet/OAuth2ClientRegistrationRepositoryConfiguration.class]: Failed to instantiate [org.springframework.security.oauth2.client.registration.InMemoryClientRegistrationRepository]: Factory method 'clientRegistrationRepository' threw exception with message: authorizationGrantType cannot be null
`

I'm using the docker-compose file to start

```
docker-compose --profile debug --profile openvsx up
```

Any idea what I am missing?

I have set authorization-grant-type in my application.yml file:

```
spring:
  security:
    oauth2:
      client:
        registration:
          gitlab:
            client-id: [redacted]
            client-secret: [redacted]
            authorization-grant-type: authorization_code
            redirect-uri: http://[webui-host]:3000/login/oauth2/code/gitlab
            scope: read_user
            clientName: OpenVSX
        provider:
          gitlab:
            authorization-uri: https://[gitlab-host]/oauth/authorize
            token-uri: https://[gitlab-host]/oauth/token
            user-info-uri: https://[gitlab-host]/api/v4/user
            jwk-set-uri: https://[gitlab-host]/oauth/discovery/keys
            user-name-attribute: id

  oauth2:
    provider: gitlab
    attribute-names:
      gitlab:
        avatar-url: avatar_url
        email: email
        full-name: name
        login-name: username
        provider-url: ''
```

HeikoBoettger-KarlStorz commented:
@amvanbaren Thank you for merging and bringing this into the master.

HeikoBoettger commented:
Hi Nikolaj,

 

I saw that there wasn’t yet an answer. Unless you just missed copying the right parts, the second part doesn’t work when put under spring, it must be under ovsx. Otherwise the question is whether you have actually changed the scripts/generate-properties.sh script or the compose-file because as far as I know that would generate and overwrite the application-property file to be used on every start.

 

Since this is merged now, have you checked with latest release?

 

Von: Nikolaj Fogh ***@***.***> 
Gesendet: Montag, 10. März 2025 11:26
An: eclipse/openvsx ***@***.***>
Cc: HeikoBoettger ***@***.***>; Mention ***@***.***>
Betreff: Re: [eclipse/openvsx] server: refactor: introduce `AuthUser` interface (PR #838)

 

Hi. I'm trying to get this branch to work for a local deployment (very excited about it - thanks for the work). But I'm getting an error when launching the server:

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'extensionService' defined in file [/app/build/classes/java/main/org/eclipse/openvsx/ExtensionService.class]: Unsatisfied dependency expressed through constructor parameter 3: Error creating bean with name 'publishExtensionVersionHandler' defined in file [/app/build/classes/java/main/org/eclipse/openvsx/publish/PublishExtensionVersionHandler.class]: Unsatisfied dependency expressed through constructor parameter 5: Error creating bean with name 'userService' defined in file [/app/build/classes/java/main/org/eclipse/openvsx/UserService.class]: Unsatisfied dependency expressed through constructor parameter 5: Error creating bean with name 'clientRegistrationRepository' defined in class path resource [org/springframework/boot/autoconfigure/security/oauth2/client/servlet/OAuth2ClientRegistrationRepositoryConfiguration.class]: Failed to instantiate [org.springframework.security.oauth2.client.registration.InMemoryClientRegistrationRepository]: Factory method 'clientRegistrationRepository' threw exception with message: authorizationGrantType cannot be null

I'm using the docker-compose file to start

docker-compose --profile debug --profile openvsx up

Any idea what I am missing?

I have set authorization-grant-type in my application.yml file:

spring:
  security:
    oauth2:
      client:
        registration:
          gitlab:
            client-id: [redacted]
            client-secret: [redacted]
            authorization-grant-type: authorization_code
            redirect-uri: http://[webui-host]:3000/login/oauth2/code/gitlab
            scope: read_user
            clientName: OpenVSX
        provider:
          gitlab:
            authorization-uri: https://[gitlab-host]/oauth/authorize
            token-uri: https://[gitlab-host]/oauth/token
            user-info-uri: https://[gitlab-host]/api/v4/user
            jwk-set-uri: https://[gitlab-host]/oauth/discovery/keys
            user-name-attribute: id
 
  oauth2:
    provider: gitlab
    attribute-names:
      gitlab:
        avatar-url: avatar_url
        email: email
        full-name: name
        login-name: username
        provider-url: ''

—
Reply to this email directly, view it on GitHub <https://github.com/eclipse/openvsx/pull/838#issuecomment-2710096807> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AFA3BRRA33AOOJLJNH5GJAD2TVSDJAVCNFSM6AAAAABTEVL2QKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMJQGA4TMOBQG4> .
You are receiving this because you were mentioned.  <https://github.com/notifications/beacon/AFA3BRWRTI76FUK2Y55F2GD2TVSDJA5CNFSM6AAAAABTEVL2QKWGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTVBRDF2O.gif> Message ID: ***@***.*** ***@***.***> >

  <https://avatars.githubusercontent.com/u/17196968?s=20&v=4> nfogh left a comment (eclipse/openvsx#838) <https://github.com/eclipse/openvsx/pull/838#issuecomment-2710096807> 

Hi. I'm trying to get this branch to work for a local deployment (very excited about it - thanks for the work). But I'm getting an error when launching the server:

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'extensionService' defined in file [/app/build/classes/java/main/org/eclipse/openvsx/ExtensionService.class]: Unsatisfied dependency expressed through constructor parameter 3: Error creating bean with name 'publishExtensionVersionHandler' defined in file [/app/build/classes/java/main/org/eclipse/openvsx/publish/PublishExtensionVersionHandler.class]: Unsatisfied dependency expressed through constructor parameter 5: Error creating bean with name 'userService' defined in file [/app/build/classes/java/main/org/eclipse/openvsx/UserService.class]: Unsatisfied dependency expressed through constructor parameter 5: Error creating bean with name 'clientRegistrationRepository' defined in class path resource [org/springframework/boot/autoconfigure/security/oauth2/client/servlet/OAuth2ClientRegistrationRepositoryConfiguration.class]: Failed to instantiate [org.springframework.security.oauth2.client.registration.InMemoryClientRegistrationRepository]: Factory method 'clientRegistrationRepository' threw exception with message: authorizationGrantType cannot be null

I'm using the docker-compose file to start

docker-compose --profile debug --profile openvsx up

Any idea what I am missing?

I have set authorization-grant-type in my application.yml file:

spring:
  security:
    oauth2:
      client:
        registration:
          gitlab:
            client-id: [redacted]
            client-secret: [redacted]
            authorization-grant-type: authorization_code
            redirect-uri: http://[webui-host]:3000/login/oauth2/code/gitlab
            scope: read_user
            clientName: OpenVSX
        provider:
          gitlab:
            authorization-uri: https://[gitlab-host]/oauth/authorize
            token-uri: https://[gitlab-host]/oauth/token
            user-info-uri: https://[gitlab-host]/api/v4/user
            jwk-set-uri: https://[gitlab-host]/oauth/discovery/keys
            user-name-attribute: id
 
  oauth2:
    provider: gitlab
    attribute-names:
      gitlab:
        avatar-url: avatar_url
        email: email
        full-name: name
        login-name: username
        provider-url: ''

—
Reply to this email directly, view it on GitHub <https://github.com/eclipse/openvsx/pull/838#issuecomment-2710096807> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AFA3BRRA33AOOJLJNH5GJAD2TVSDJAVCNFSM6AAAAABTEVL2QKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMJQGA4TMOBQG4> .
You are receiving this because you were mentioned.  <https://github.com/notifications/beacon/AFA3BRWRTI76FUK2Y55F2GD2TVSDJA5CNFSM6AAAAABTEVL2QKWGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTVBRDF2O.gif> Message ID: ***@***.*** ***@***.***> >

nfogh commented:
> Hi Nikolaj, I saw that there wasn’t yet an answer. Unless you just missed copying the right parts, the second part doesn’t work when put under spring, it must be under ovsx. Otherwise the question is whether you have actually changed the scripts/generate-properties.sh script or the compose-file because as far as I know that would generate and overwrite the application-property file to be used on every start. Since this is merged now, have you checked with latest release? Von: Nikolaj Fogh ***@***.***> Gesendet: Montag, 10. März 2025 11:26 An: eclipse/openvsx ***@***.***> Cc: HeikoBoettger ***@***.***>; Mention ***@***.***> Betreff: Re: [eclipse/openvsx] server: refactor: introduce `AuthUser` interface (PR #838) Hi. I'm trying to get this branch to work for a local deployment (very excited about it - thanks for the work). But I'm getting an error when launching the server: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'extensionService' defined in file [/app/build/classes/java/main/org/eclipse/openvsx/ExtensionService.class]: Unsatisfied dependency expressed through constructor parameter 3: Error creating bean with name 'publishExtensionVersionHandler' defined in file [/app/build/classes/java/main/org/eclipse/openvsx/publish/PublishExtensionVersionHandler.class]: Unsatisfied dependency expressed through constructor parameter 5: Error creating bean with name 'userService' defined in file [/app/build/classes/java/main/org/eclipse/openvsx/UserService.class]: Unsatisfied dependency expressed through constructor parameter 5: Error creating bean with name 'clientRegistrationRepository' defined in class path resource [org/springframework/boot/autoconfigure/security/oauth2/client/servlet/OAuth2ClientRegistrationRepositoryConfiguration.class]: Failed to instantiate [org.springframework.security.oauth2.client.registration.InMemoryClientRegistrationRepository]: Factory method 'clientRegistrationRepository' threw exception with message: authorizationGrantType cannot be null I'm using the docker-compose file to start docker-compose --profile debug --profile openvsx up Any idea what I am missing? I have set authorization-grant-type in my application.yml file: spring: security: oauth2: client: registration: gitlab: client-id: [redacted] client-secret: [redacted] authorization-grant-type: authorization_code redirect-uri: http://[webui-host]:3000/login/oauth2/code/gitlab scope: read_user clientName: OpenVSX provider: gitlab: authorization-uri: https://[gitlab-host]/oauth/authorize token-uri: https://[gitlab-host]/oauth/token user-info-uri: https://[gitlab-host]/api/v4/user jwk-set-uri: https://[gitlab-host]/oauth/discovery/keys user-name-attribute: id oauth2: provider: gitlab attribute-names: gitlab: avatar-url: avatar_url email: email full-name: name login-name: username provider-url: '' — Reply to this email directly, view it on GitHub <[#838 (comment)](https://github.com/eclipse/openvsx/pull/838#issuecomment-2710096807)> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AFA3BRRA33AOOJLJNH5GJAD2TVSDJAVCNFSM6AAAAABTEVL2QKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMJQGA4TMOBQG4> . You are receiving this because you were mentioned. <https://github.com/notifications/beacon/AFA3BRWRTI76FUK2Y55F2GD2TVSDJA5CNFSM6AAAAABTEVL2QKWGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTVBRDF2O.gif> Message ID: ***@***.*** ***@***.***> > <https://avatars.githubusercontent.com/u/17196968?s=20&v=4> nfogh left a comment ([eclipse/openvsx#838](https://github.com/eclipse/openvsx/pull/838)) <[#838 (comment)](https://github.com/eclipse/openvsx/pull/838#issuecomment-2710096807)> Hi. I'm trying to get this branch to work for a local deployment (very excited about it - thanks for the work). But I'm getting an error when launching the server: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'extensionService' defined in file [/app/build/classes/java/main/org/eclipse/openvsx/ExtensionService.class]: Unsatisfied dependency expressed through constructor parameter 3: Error creating bean with name 'publishExtensionVersionHandler' defined in file [/app/build/classes/java/main/org/eclipse/openvsx/publish/PublishExtensionVersionHandler.class]: Unsatisfied dependency expressed through constructor parameter 5: Error creating bean with name 'userService' defined in file [/app/build/classes/java/main/org/eclipse/openvsx/UserService.class]: Unsatisfied dependency expressed through constructor parameter 5: Error creating bean with name 'clientRegistrationRepository' defined in class path resource [org/springframework/boot/autoconfigure/security/oauth2/client/servlet/OAuth2ClientRegistrationRepositoryConfiguration.class]: Failed to instantiate [org.springframework.security.oauth2.client.registration.InMemoryClientRegistrationRepository]: Factory method 'clientRegistrationRepository' threw exception with message: authorizationGrantType cannot be null I'm using the docker-compose file to start docker-compose --profile debug --profile openvsx up Any idea what I am missing? I have set authorization-grant-type in my application.yml file: spring: security: oauth2: client: registration: gitlab: client-id: [redacted] client-secret: [redacted] authorization-grant-type: authorization_code redirect-uri: http://[webui-host]:3000/login/oauth2/code/gitlab scope: read_user clientName: OpenVSX provider: gitlab: authorization-uri: https://[gitlab-host]/oauth/authorize token-uri: https://[gitlab-host]/oauth/token user-info-uri: https://[gitlab-host]/api/v4/user jwk-set-uri: https://[gitlab-host]/oauth/discovery/keys user-name-attribute: id oauth2: provider: gitlab attribute-names: gitlab: avatar-url: avatar_url email: email full-name: name login-name: username provider-url: '' — Reply to this email directly, view it on GitHub <[#838 (comment)](https://github.com/eclipse/openvsx/pull/838#issuecomment-2710096807)> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AFA3BRRA33AOOJLJNH5GJAD2TVSDJAVCNFSM6AAAAABTEVL2QKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMJQGA4TMOBQG4> . You are receiving this because you were mentioned. <https://github.com/notifications/beacon/AFA3BRWRTI76FUK2Y55F2GD2TVSDJA5CNFSM6AAAAABTEVL2QKWGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTVBRDF2O.gif> Message ID: ***@***.*** ***@***.***> >

Hi

I tried with the latest release, and it works great there. It might have had something to do with the scripts you refer to.

Thanks for following up on my message. Since the MR was closed, i didn't think anyone was paying attention :)
